src/frequenz/sdk/timeseries/battery_pool/_methods.py

```diff
@@ -0,0 +1,413 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Methods for processing battery-inverter data."""
+import asyncio
+import logging
+from abc import ABC, abstractmethod
+from datetime import datetime, timedelta, timezone
+from typing import Dict, Generic, Optional, Set
+
+from frequenz.channels import Broadcast, Receiver
+
+from ..._internal._constants import (
+    MAX_BATTERY_DATA_AGE_SEC,
+    WAIT_FOR_COMPONENT_DATA_SEC,
+)
+from ..._internal.asyncio import cancel_and_await
+from ._component_metrics import ComponentMetricsData
+from ._metric_calculator import MetricCalculator, T
+from ._component_metric_fetcher import (
+    LatestBatteryMetricsFetcher,
+    LatestInverterMetricsFetcher,
+    ComponentMetricFetcher,
+    ResampledComponentMetricFetcher,
+    ResamplerArgs,
+)
+
+_logger = logging.getLogger(__name__)
+
+# Constant needed in the code
+_MIN_TIMESTAMP = datetime.min.replace(tzinfo=timezone.utc)
+
+
+class BaseMethod(ABC, Generic[T]):
```

Comment:
Maybe `AggregateMethod` or `SendMethod` whatever this is? Without reading the code I have no idea if this is a method for sending, aggregating or what :)

Reply:
Done : changed to `AggregateMethod`